res/raw/webxdc.js

```diff
@@ -41,26 +41,74 @@ window.webxdc = (() => {
     },
 
     sendToChat: async (message) => {
-        const data = {};
-        if (!message.text && !message.file) {
-            return Promise.reject("Invalid empty message, at least one of text or file should be provided");
-        }
-        if (message.text) {
-            data.text = message.text;
+      const data = {};
+      if (!message.file && !message.text) {
+        return Promise.reject(
+          "Error from sendToChat: Invalid empty message, at least one of text or file should be provided"
+        );
+      }
+      /** @type {(file: Blob) => Promise<string>} */
+      const blob_to_base64 = (file) => {
+        const data_start = ";base64,";
+        return new Promise((resolve, reject) => {
+          const reader = new FileReader();
+          reader.readAsDataURL(file);
+          reader.onload = () => {
```

Comment:
shouldn't the readAsDataURL be called after setting the event listener? even if due to some JS weirdness this works, isn't easier to read/understand if the listener is set before the event is triggered?